### DISCUSSION TITLE
How to see state of arguments?

### DISCUSSION THREAD

**Discussion by tombohub** [bilal-fazlani/commanddotnet#483] [Q&A] [answered]
If I run command is there a way to display state of all arguments, boolean flags etc. Like python's argsparse 

```python
>>>parser = argparse.ArgumentParser()
>>>parser.add_argument('foo', type=int)
>>>parser.set_defaults(bar=42, baz='badger')
>>>parser.parse_args(['736'])
Namespace(bar=42, baz='badger', foo=736)
```

**Comment by drewburlingame** [comment, marked as answer]
There's the [parse directive](https://commanddotnet.bilal-fazlani.com/diagnostics/parse-directive/) and [command logger](https://commanddotnet.bilal-fazlani.com/diagnostics/command-logger/). 

Both of these can display for the end-user.  If they aren't appropriate, can you explain your use case more?  The code for these may also be helpful in understanding how to access what you're looking for.
